src/instrumenter/state_var_instrumenter.ts

```diff
@@ -276,12 +274,22 @@ function getWrapperName(
             }
             const [expectedTyp, expr] = el;
 
-            const exprT = getExprSType(expr, astTypeNameToSType(expectedTyp));
+            const exprT = getMaterialExprType(
+                expr,
+                version,
+                updateNode,
+                typeNameToTypeNode(expectedTyp)
+            );
             return `idx_${getTypeDescriptor(exprT)}`;
         })
         .join("_");
+
     const additionalArgsString = additionalArgs
-        .map(([expr, typ]) => getTypeDescriptor(getExprSType(expr, astTypeNameToSType(typ))))
```

Comment:
A: Because otherwise we have to assume some default values for storage location for complex key types (bytes/strings in mappigns) and additional arguments (values we are assigning). If we assume default types we may introduce extra deep-copies in the code. The tradeoff we are picking here is to emit multiple specialized versions of the same wrapper depending on arguements. Also this way we can avoid dealing with implicit casts from array literals to more general storage arrays.